src/ray/intersect_default.rs

```diff
@@ -0,0 +1,47 @@
+use super::Ray;
+use crate::{aabb::Aabb, utils::fast_max};
+use nalgebra::{ClosedMul, ClosedSub, Scalar, SimdPartialOrd};
+use num::Zero;
+
+pub trait RayIntersection<T, const D: usize>
+where
+    T: Copy + Scalar,
+{
+    fn ray_intersects_aabb(&self, aabb: &Aabb<T, D>) -> bool;
+}
+
+#[cfg(not(feature = "simd"))]
+impl<T, const D: usize> RayIntersection<T, D> for Ray<T, D>
+where
+    T: Scalar + Copy + ClosedSub + ClosedMul + Zero + PartialOrd + SimdPartialOrd,
+{
+    fn ray_intersects_aabb(&self, aabb: &Aabb<T, D>) -> bool {
+        let lbr = (aabb[0].coords - self.origin.coords).component_mul(&self.inv_direction);
+        let rtr = (aabb[1].coords - self.origin.coords).component_mul(&self.inv_direction);
+
+        let (inf, sup) = lbr.inf_sup(&rtr);
+
+        let tmin = inf.max();
+        let tmax = sup.min();
+
+        tmax > fast_max(tmin, T::zero())
+    }
+}
+
+#[cfg(all(feature = "simd", target_arch = "x86_64"))]
```

Comment:
How about removing the feature check here and moving this function into `intersect_x86_64.rs`? Then you could just slap a feature check on top of the `mod` lines and I think it'd be a bit cleaner.

Reply:
I was thinking I might add some explicit SIMD for ARM as well, which is why I left it in default. It is also the default implementation which I thought made sense for it to be here. Its unfortunate that the only difference between the two is the `default` keyword.

Reply:
Sure and I think that's fine. I was just commenting about code order unless I'm missing something. I'm thinking we should rather have
```rust
#[cfg(not(feature = "simd"))]
mod default_impl;

#[cfg(all(feature = "simd", target_arch = "x86_64"))]
mod x86_64_impl;

#[cfg(all(feature = "simd", target_arch = "aarch64"))]
mod aarch64_impl;
```

as opposed to `cfg`-ing away the specific implementations per file.

Reply:
I believe the ones in the `default` file are the only ones that are per function (and there are two).

The x86_64 file is completely guarded like you posted above already.